README.md

```diff
@@ -80,71 +86,137 @@ module "ecs_fargate_app" {
-              "options": {
-                "awslogs-group": "${module.ecs_fargate_app.log_group_path}",
-                "awslogs-region": "${data.aws_region.current.name}",
-                "awslogs-stream-prefix": "ecs"
-                }
+    #NOTE: ALBs are not created by the module.
+    health_check_grace_period_seconds = 600
+    lb_listener_prod_arn              = "arn:aws:elasticloadbalancing:us-east-1:123456789876:listener/app/my_prd_listener/ed9abd4ebab2925a/9176f3bfebd6457f"
+    lb_listener_test_arn              = "arn:aws:elasticloadbalancing:us-east-1:123456789876:listener/app/my_test_listener/j6d77m8jttgd4g853/mqo39m4lcj3lfk3"
+    lb_target_group_blue_arn          = "arn:aws:elasticloadbalancing:us-east-1:123456789876:targetgroup/my_blue_target_group/f1fec68432dd54c0"
+    lb_target_group_blue_name         = my_blue_target_group
+    lb_target_group_green_name        = my_green_target_group
+    lb_container_name                 = "containername" # has to match name in container definition within task_definition
+    lb_container_port                 = 8080       # has to match port in container definition within task_definition
+
+    codedeploy_role_arn              = "arn:aws:iam::123456789876:role/my_codedeploy_role"
+    codedeploy_termination_wait_time = "5"
+    codebuild_auto_rollback_enabled  = true
+    codebuild_auto_rollback_events   = ["DEPLOYMENT_FAILURE"]
+
+    codepipeline_role_arn        = "arn:aws:iam::123456789876:role/my_codepipeline_role"
+    codepipeline_source_bucket_id  = "my_codepipeline_source_bucket_name"
+    codepipeline_source_object_key = "deployment/ecs/${var.application_name}-artifacts.zip"
+
+    container_repo_name         = "my_ecr_repository"
+    container_target_tag        = "latest" #
+    container_duplicate_targets = "${var.name_prefix}-${var.application_name}-ecr-repo-prd"
+
+    deployment_manual_approval  = local.ecs_deployment_approval[var.env_name] // boolean for environment to deploy into (prd)
+    duplication_manual_approval = local.ecs_duplication_approval[var.env_name] // boolen for environment to duplicate from (dev)
+
+    use_custom_capacity_provider_strategy = true //if false, custom_capacity_provider_strategy needs to be an emtpy block = {}
+    custom_capacity_provider_strategy = {
+      primary_capacity_provider_base      = 1
+      primary_capacity_provider           = "FARGATE"
+      primary_capacity_provider_weight    = 10
+      secondary_capacity_provider         = "FARGATE_SPOT"
+      secondary_capacity_provider_weight  = 1
+    }
+
+    taskdef_family             = "MyService"
+    taskdef_execution_role_arn = "arn:aws:iam::123456789876:role/my_taskdef_role"
+    taskdef_task_role_arn      = "arn:aws:iam::123456789876:role/my_taskdef_role"
+    taskdef_network_mode       = "awsvpc"
+    taskdef_requires_compatibilities = [
+      "FARGATE"
+    ]
+    taskdef_cpu    = 2048
+    taskdef_memory = 4096
+
+    taskdef_container_definitions = <<-TASKDEF
```

Comment:
Nit: Consider using HCL in the clients and `jsonencode` in the module which makes for a cleaner client experience and stronger typing enforcement.

Reply:
I did - jsonencode() destroys the "<" and ">" in the "<IMAGE1_NAME>" string, which breaks the deployment functionality (sadface).  Are you aware of a way around this problem?

Reply:
Maybe this comment helps: https://github.com/hashicorp/terraform/pull/18871#issuecomment-422220699

Reply:
This, indeed, is a *much* better approach and will be used in the next revision (coming soon!).